ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java

```diff
@@ -212,21 +178,52 @@ public APIGatewayProxyResponseEvent handleRequest(
                                     ipvAuthorisationService.validateResponse(
                                             input.getQueryStringParameters(),
                                             session.getSessionId()));
-            if (errorObject.isPresent()) {
-                return generateAuthenticationErrorResponse(
-                        authRequest,
-                        new ErrorObject(ACCESS_DENIED_CODE, errorObject.get().getDescription()),
-                        false,
-                        clientSessionId,
-                        session.getSessionId());
-            }
             var userProfile =
                     dynamoService
                             .getUserProfileFromEmail(session.getEmailAddress())
                             .orElseThrow(
                                     () ->
                                             new IpvCallbackException(
                                                     "Email from session does not have a user profile"));
+            var rpPairwiseSubject =
+                    segmentedFunctionCall(
+                            "calculatePairwiseSubject",
+                            () ->
+                                    ClientSubjectHelper.getSubject(
+                                            userProfile,
+                                            clientRegistry,
+                                            dynamoService,
+                                            configurationService.getInternalSectorUri()));
+
+            var internalPairwiseSubjectId =
+                    ClientSubjectHelper.calculatePairwiseIdentifier(
+                            userProfile.getSubjectID(),
+                            URI.create(configurationService.getInternalSectorUri()),
+                            dynamoService.getOrGenerateSalt(userProfile));
+
+            if (errorObject.isPresent()) {
+                var accountInterventionStatus =
+                        ipvCallbackHelper.getAccountInterventionStatus(internalPairwiseSubjectId);
+                if (configurationService.isAccountInterventionServiceEnabled()) {
+                    if (accountInterventionStatus.blocked()) {
+                        // TODO: back channel logout + redirect to blocked page
```

Comment:
Are these TODOs needed still?

Reply:
It's not a TODO for this ticket, more of a placeholder to flag that the functionality here is incomplete and this is what needs to be done